replay2/memory_remapper/memory_remapper.cc

```diff
@@ -14,4 +14,107 @@
 
 #include "memory_remapper.h"
 
-void foo(int var) { return; }
+namespace agi {
+namespace replay2 {
+
+	void MarkDeadAddressRange(const ReplayAddressRange& replayAddressRange) {
```

Comment:
I was thinking more like a macro that simply memsets a memory so that it can be used from anywhere without repeating this implementation. And looping seems unnecessary when it can easily memset in one pass and one line

Reply:
My concern here is more like if we try to do same thing with compiler and if we have a bug, we can mark a non-free memory as free, so the only issue I have is if it causes confusion. Because AFAIK this is a common behavior. If you think that we wouldn't have a clash that's fine. I am just acting on the fact that in the past a lot of thing reinvented in AGI and I am kinda a bit conscious about it. Decision is up to you.